### PR TITLE
Add support for SET … GET option when version >=6.2.0 -resolves #2264

### LINKED ISSUE #2264: Add support for SET … GET option
Redis 6.2 supports a GET option to return the old value stored at key, or nil when key did not exist. See [https://redis.io/commands/set](https://redis.io/commands/set)

### PR DESCRIPTION
**Pull request by Jayice-zjw**
added support for SET ...GET option
See https://redis.io/commands/set
resolves #2264

### CODE REVIEW

**Comment by sazzad16** [comment]
@Jayice-zjw Please add test codes.

**Comment by Jayice-zjw** [comment]
> @Jayice-zjw Please add test codes.

done

File: src/main/java/redis/clients/jedis/params/SetParams.java [hunk 1]
```diff
@@ -69,6 +79,9 @@ public SetParams xx() {
     if (contains(XX)) {
       byteParams.add(SafeEncoder.encode(XX));
     }
+    if (contains(GET)) {
+      byteParams.add(SafeEncoder.encode(GET));
+    }
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/params/SetParams.java:84]: Move this `if` block after all `if` blocks.

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/params/SetParams.java (hunk 1 above)
Comment (Jayice-zjw) [reply src/main/java/redis/clients/jedis/params/SetParams.java]: done.
btw, does the order of the if block here need to follow the order supported by the version?

**Review by sazzad16 — APPROVED** [review]

File: src/main/java/redis/clients/jedis/params/SetParams.java (hunk 1 above)
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/params/SetParams.java]: not necessarily. just being safe here

**Comment by sazzad16** [comment]
@Jayice-zjw I'd appreciate if you could also test `nil` when the key is absent.

**Comment by Jayice-zjw** [comment]
> @Jayice-zjw I'd appreciate if you could also test `nil` when the key is absent.

all right.added

**Review by gkorland — APPROVED** [review]
